### PR TITLE
workflows/release: use OIDC exchange

### PR DESCRIPTION
**Pull request by woodruffw**
Now that we've merged this upstream 🙂

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,7 +38,6 @@ jobs:
       uses: pypa/gh-action-pypi-publish@release/v1
       with:
         user: __token__
-        password: ${{ secrets.PYPI_TOKEN }}
```
Comment (di) [inline .github/workflows/release.yml]: We can remove all of these lines, right?

Comment (woodruffw) [reply .github/workflows/release.yml]: Yep! I missed that last one; just removed.

**Comment by woodruffw** [comment]
I've also deleted the `PYPI_TOKEN` secret. @di I think you'll have to remove the corresponding token on the PyPI side, since it's not tied to my account (AFAICT).

**Comment by di** [comment]
@woodruffw I'm not seeing it on my account either.

**Comment by woodruffw** [comment]
Yeah, my mistake -- it was named poorly on my side. It's removed now!

**Pull request merged**